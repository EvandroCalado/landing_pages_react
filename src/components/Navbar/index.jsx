import P from 'prop-types';
import * as Styled from './styles';
import MenuLink from '../MenuLink';

const Navbar = ({ links = [] }) => {
  return (
    <Styled.Container>
      {links.map((link) => (
        <MenuLink key={link.link} {...link} />
      ))}
    </Styled.Container>
  );
};

Navbar.propTypes = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    })
  ),
};

export default Navbar;
