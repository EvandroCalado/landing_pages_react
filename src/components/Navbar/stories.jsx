import Navbar from '.';
import {mock} from "./mock"

const NavbarStories = {
  title: 'Navbar',
  component: Navbar,
  args: {
    links: mock,
  },
};

export const Template = (args) => {
  return (
    <div>
      <Navbar {...args} />
    </div>
  );
};

export default NavbarStories;