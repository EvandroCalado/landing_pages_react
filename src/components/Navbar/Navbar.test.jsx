import Navbar from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render.theme';

describe('<Navbar />', () => {
  it('should render', () => {
    renderTheme(<Navbar>Children</Navbar>);
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});