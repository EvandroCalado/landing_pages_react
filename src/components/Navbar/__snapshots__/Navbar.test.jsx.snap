// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match snapshot 1`] = `
Array [
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 1"
    target="_self"
  >
    link 1
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 2"
    target="_self"
  >
    link 2
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 3"
    target="_self"
  >
    link 3
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 4"
    target="_self"
  >
    link 4
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 5"
    target="_self"
  >
    link 5
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 6"
    target="_self"
  >
    link 6
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 7"
    target="_self"
  >
    link 7
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 8"
    target="_self"
  >
    link 8
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 9"
    target="_self"
  >
    link 9
  </a>,
  .c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target 10"
    target="_self"
  >
    link 10
  </a>,
]
`;
